src/components/App/App.component.jsx

```diff
@@ -1,57 +1,70 @@
-import Layout from '../Layout';
-import { random } from '../../utils/fns';
+import React, { useState } from 'react';
+import HomePage from '../../pages/Home/Home.page';
+import NavBar from '../NavBar';
+import useVideoApi from '../../pages/Home/useVideosAPI';
+import DetailsView from '../DetailsView';
 
 function App() {
-  useLayoutEffect(() => {
-    const { body } = document;
+  const { videos, loading, error, fetchVideos } = useVideoApi();
+  const [searchField, setSearchField] = useState('');
+  const [displayDetailView, setDisplayDetailView] = useState(false);
+  const [videotoPlay, setVideotoPlay] = useState({
+    url: '',
+    title: '',
+    description: '',
+  });
+
+  const handleSearchField = (event) => {
+    setSearchField(event.target.value);
+  };
 
-    function rotateBackground() {
-      const xPercent = random(100);
-      const yPercent = random(100);
-      body.style.setProperty('--bg-position', `${xPercent}% ${yPercent}%`);
-    }
+  const handleSearchButton = () => {
+    fetchVideos(searchField);
+  };
 
-    const intervalId = setInterval(rotateBackground, 3000);
-    body.addEventListener('click', rotateBackground);
+  const showDetailView = (url, title, description) => {
+    setVideotoPlay({ url: url, title: title, description: description });
+    setDisplayDetailView(true);
+  };
 
-    return () => {
-      clearInterval(intervalId);
-      body.removeEventListener('click', rotateBackground);
-    };
-  }, []);
+  if (displayDetailView) {
+    return (
+      <div>
+        {/** Header */}
+        <header>
+          <NavBar.LightNavBar
+            handleSearchField={handleSearchField}
+            handleSearchButton={handleSearchButton}
+          />
+        </header>
+        {/** Page main content */}
+        <main>
+          <DetailsView video={videotoPlay} relatedVideos={videos} />
+        </main>
+      </div>
+    );
+  }
 
   return (
-    <BrowserRouter>
-      <AuthProvider>
-        <Layout>
-          <Switch>
-            <Route exact path="/">
-              <HomePage />
-            </Route>
-            <Route exact path="/login">
-              <LoginPage />
-            </Route>
-            <Private exact path="/secret">
-              <SecretPage />
-            </Private>
-            <Route path="*">
-              <NotFound />
-            </Route>
-          </Switch>
-          <Fortune />
-        </Layout>
-      </AuthProvider>
-    </BrowserRouter>
+    <div>
+      {/** Header */}
+      <header>
+        <NavBar.LightNavBar
+          handleSearchField={handleSearchField}
+          handleSearchButton={handleSearchButton}
+        />
+      </header>
+      {/** Page main content */}
+      <main>
+        {!loading && videos ? (
+          <HomePage videos={videos} showDetailView={showDetailView} />
+        ) : !loading && error ? (
+          <div>An error has ocurred</div>
+        ) : (
+          <div>Loading...</div>
+        )}
+      </main>
+    </div>
```

Comment:
Nice state app cases: loading, error and results.

Using nested ternary operation makes hard to read the conditional rendering.

You could refactor with something like this:

```js
 <main>
    {loading && <div>Loading...</div>}
    
    {!loading && videos && <HomePage videos={videos} showDetailView={showDetailView} />}

   {!loading && ! error && <div>An error has ocurred</div>}
 </main>
```

Which could be refactored as well like:

```js
const isContent = !loading && videos
const isError = !loading && error

rest of the code....

```js
 <main>
    {loading && <div>Loading...</div>}
    
    {isContent && <HomePage videos={videos} showDetailView={showDetailView} />}

   {isError && <div>An error has ocurred</div>}
 </main>
```


```